### PR TITLE
[Merged by Bors] - Path

### PR DESCRIPTION
**Pull request by Muqito**
Added some suggestions of code comments to clear up what the code is actually trying to do. Still some things I could improve though; like what the q! and the shortcut for smart_quote actually yields in the end.

### CODE REVIEW

File: macros/src/path.rs [hunk 1]
```diff
@@ -30,35 +30,38 @@ pub fn compile(
     let path = path.value();
     assert!(path.starts_with('/'), "Path should start with /");
 
-    let segments = path.split('/');
-    let len = segments.clone().filter(|&s| s != "").count();
-
-    if len == 0 {
+    // It doesn't make any sense to have a route containing //
+    if path.find("//").is_some() {
```
Comment (kdy1) [inline macros/src/path.rs]: I think we should panic with a helpful error message.

**Comment by Muqito** [comment]
@kdy1 Is this more of what you had in mind? :)

**Comment by Muqito** [comment]
Wait; I'll add a test to it as well

**Comment by Muqito** [comment]
I'll add some test cases as well

**Comment by kdy1** [comment]
@Muqito Yes. Thanks.
I'll mege it after adding some tets

**Comment by Muqito** [comment]
@kdy1

**Pull request closed without merging**